### ISSUE TITLE
This vulnerability is a UAF (Use-After-Free) vulnerability discovered in the file /fluent-bit/plugins/custom_calyptia/calyptia.c.

### ISSUE THREAD

**Issue by LuMingYinDetect** [fluent/fluent-bit#8530]
Hello, I am a graduate student specializing in static analysis of programs. Recently, while using a static analysis tool to detect issues in open-source projects, I found several defects in the project. The description of the defects can be found at the following link:https://github.com/LuMingYinDetect/fluent-bit_defects/blob/main/fluent-bit_detect_1.md

**Comment by patrick-stephens** [comment]
Can you include the details here rather than in an uncontrolled external repo?

Please submit a PR to resolve if possible.

**Comment by LuMingYinDetect** [comment]
> Can you include the details here rather than in an uncontrolled external repo?
> 
> Please submit a PR to resolve if possible.


Hello! I am the discoverer of this CVE. First of all, I apologize for not contacting you in advance because this is my first time applying for a CVE, and I may not be familiar enough with the application process. This time, I found two issues in the software in total, and I submitted an issue two days ago. Here are our discussion links: https://github.com/fluent/fluent-bit/issues/8531、https://github.com/fluent/fluent-bit/issues/8530. The one you mentioned seems to be an external library according to my discussion with the developers. The other defect appears to be a double-free defect in the software. We are considering how to fix it, and I have already submitted a PR for your reference!
